src/config.js

```diff
@@ -0,0 +1,312 @@
+
+  logger.debug(
+    `Searching for ${CONFIG_FILE_NAME} files from ${currentDir} to ${homeDirResolved}`,
+  );
+
+  // Collect all directories from cwd to home
+  const directories = [];
+  while (true) {
+    directories.push(currentDir);
+
+    // Stop if we've reached the home directory
+    if (currentDir === homeDirResolved) {
+      break;
+    }
+
+    // Stop if we've reached the root directory
+    const parentDir = deps.dirname(currentDir);
+    if (parentDir === currentDir) {
+      break;
+    }
+
+    currentDir = parentDir;
+  }
+
+  // Read configs in reverse order (from home to cwd)
+  // so that configs array is ordered from least specific to most specific
+  for (let i = directories.length - 1; i >= 0; i--) {
+    const dir = directories[i];
+    const configPath = deps.join(dir, CONFIG_FILE_NAME);
+    const config = await readConfigFile(configPath);
+    if (config) {
+      configs.push(config);
+    }
+  }
+
+  return configs;
+};
+
+/**
+ * Merges environment configurations from multiple config files
+ * Returns merged environment or null if not found
+ */
+const mergeEnvironmentConfigs = (configs, envName) => {
+  const merged = {};
+
+  for (const config of configs) {
+    if (
+      config.environments &&
+      config.environments[envName] &&
+      typeof config.environments[envName] === "object"
+    ) {
+      for (const [key, value] of Object.entries(config.environments[envName])) {
+        if (key === "inputVariables") {
+          if (!merged.inputVariables) {
+            merged.inputVariables = {};
+          }
+          Object.assign(merged.inputVariables, value);
+        } else if (key === "secrets") {
+          merged.secrets = value;
+        } else {
+          merged[key] = value;
+        }
+      }
+    }
+  }
+
+  return Object.keys(merged).length > 0 ? merged : null;
+};
+
+/**
+ * Merges defaults from config objects
+ * Later configs (closer to cwd) override earlier configs (closer to home)
+ */
+const mergeDefaultConfigs = (configs) => {
+  const merged = {};
+
+  for (const config of configs) {
+    if (config.defaults && typeof config.defaults === "object") {
+      for (const [key, value] of Object.entries(config.defaults)) {
+        if (key === "inputVariables") {
+          if (!merged.inputVariables) {
+            merged.inputVariables = {};
+          }
+          Object.assign(merged.inputVariables, value);
+        } else if (key === "secrets") {
+          merged.secrets = value;
+        } else {
+          merged[key] = value;
+        }
+      }
+    }
+  }
+
+  return merged;
+};
+
+/**
+ * Applies config to command-line arguments
+ * CLI arguments always take precedence over config
+ */
+const applyConfigToArgs = async (args, configData) => {
+  if (!configData) {
+    return args;
+  }
+
+  const enrichedArgs = { ...args };
+
+  const argMapping = {
+    environment: "environment",
+    headers: "headers",
+    inputVariables: "inputVariables",
```

Comment:
there is inconsistency, followRedirects: follow-redirects. Let's try to fix this..